File: aon-occam/src/main/java/com/esferalia/aon/occam/api/model/registry/RegistryItemStatus.java
package com.esferalia.aon.occam.api.model.registry;

public enum RegistryItemStatus {
	ACTIVE,
	INTERESTED,
	REFUSED,
	INACTIVE;
	
	private RegistryItemStatus() {
		
	}
	
	public byte value() {
		return (byte) this.ordinal();
	}
	
	public static RegistryItemStatus safeValueOf( Byte i ) {
		if (i == null) return null;
		return safeValueOf( i.intValue() ); 
	}
	public static RegistryItemStatus safeValueOf( Integer i ) {
		if (i == null) return null;
		if (i < 0 || i >= RegistryItemStatus.values().length) return null;
		return RegistryItemStatus.values()[i];
	}
}

File: aon-solutions/src/main/webapp/models/product/RegistryItemStatus.js
export const RegistryItemStatus = {
    ACTIVE: "ACTIVE",
    INTERESTED: "INTERESTED",
    REFUSED: "REFUSED",
    INACTIVE: "INACTIVE",

    getText: function (i) {
        if (i === null) return null;

        switch (i) {
            case RegistryItemStatus.INTERESTED:
                return "INTERESADO";
            case RegistryItemStatus.REFUSED:
                return "RECHAZADO";
            case RegistryItemStatus.INACTIVE:
                return "Inactivo";
            default:
                return "ACTIVO";
        }
    },
};
